src/telegrafs/components/TelegrafConfigOverlayForm.tsx

```diff
@@ -0,0 +1,138 @@
+
+// Types
+import {AppState, ResourceType, Telegraf} from 'src/types'
+
+// Selectors
+import {getAll} from 'src/resources/selectors'
+
+interface OwnProps {
+  onClose: () => void
+  onUpdateTelegraf: (telegraf: Telegraf) => void
+}
+
+type Params = {orgID: string; id: string}
+interface Match {
+  params: Params
+}
+
+type ReduxProps = ConnectedProps<typeof connector>
+type Props = OwnProps & ReduxProps & RouteComponentProps
+
+const TelegrafConfigOverlayForm: FC<Props> = ({
+  telegrafs,
+  onUpdateTelegraf,
+  onClose,
+}) => {
+  const match: Match = useRouteMatch({
+    path: '/orgs/:orgID/load-data/telegrafs/:id/view',
+    exact: true,
+    strict: false,
+  })
+
+  let telegraf
+
+  if (match && match.params && match.params.id) {
+    telegraf = telegrafs.find(tel => tel.id === match.params.id)
+  }
+
+  const [workingConfig, updateWorkingConfig] = useState<string>(
+    telegraf?.config || ''
+  )
+
+  if (!telegraf) {
+    return null
+  }
+
+  const handleDownloadConfig = () => {
+    const {name} = telegraf
+    downloadTextFile(workingConfig, name || 'telegraf', '.conf')
+  }
+
+  const handleSaveConfig = (): void => {
+    onUpdateTelegraf({...telegraf, config: workingConfig})
+  }
+
+  const handleChangeConfig = (config: string): void => {
+    updateWorkingConfig(config)
+  }
+
+  return (
+    <>
+      <Overlay.Body>
+        <Panel style={{marginBottom: '8px'}}>
+          <Panel.Body
+            direction={FlexDirection.Row}
+            alignItems={AlignItems.Center}
+            justifyContent={JustifyContent.SpaceBetween}
+          >
+            <Heading element={HeadingElement.H5}>{telegraf.name}</Heading>
+            <Button
+              icon={IconFont.Download}
+              color={ComponentColor.Secondary}
+              text="Download Config"
+              onClick={handleDownloadConfig}
+            />
+          </Panel.Body>
+        </Panel>
+        <div className="config-overlay">
+          <TelegrafConfig
+            config={workingConfig}
+            onChangeConfig={handleChangeConfig}
+          />
+        </div>
+      </Overlay.Body>
+      <FeatureFlag name="editTelegrafs">
+        <Overlay.Footer>
+          <Button
+            color={ComponentColor.Default}
+            text="Cancel"
+            onClick={onClose}
+          />
+          <ConfirmationButton
+            color={ComponentColor.Success}
+            text="Save Changes"
+            onConfirm={handleSaveConfig}
+            confirmationButtonText="Save"
+            confirmationButtonColor={ComponentColor.Success}
+            confirmationLabel="This cannot be undone, are you sure?"
+          />
+        </Overlay.Footer>
+      </FeatureFlag>
+    </>
+  )
+}
+
+const mstp = (state: AppState) => {
+  return {
+    telegrafs: getAll<Telegraf>(state, ResourceType.Telegrafs),
+  }
+}
+
+const connector = connect(mstp)
+
+export default connector(withRouter(TelegrafConfigOverlayForm))
```

Comment:
Nit: seems like a lot of work to set this FC up when we could use hooks and make this component a bit more contained and isolated.

Nit2: Maybe instead of drilling `onClose` all the way from the `OverlayController` we can use a hook to instantiate it here?